src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2559,6 +2571,9 @@ def load_key_and_certificates_from_pkcs12(self, data, password):
         if x509_ptr[0] != self._ffi.NULL:
             x509 = self._ffi.gc(x509_ptr[0], self._lib.X509_free)
             cert = self._ossl2cert(x509)
+            maybe_name = self._lib.X509_alias_get0(x509_ptr[0], self._ffi.NULL)
```

Comment:
I asked this previously, but can there be multiple friendly names in a PKCS12 (e.g. one per additional certificate)? If yes then this gets a bit trickier because we should have an API that can return them all. You could imagine that being `(key, (name, cert), [(name, cert), ...])` I suppose, although it's worth asking ourselves if we've reached the threshold of needing a real PKCS12 object at that point.

Reply:
I think there can be, but right now it's not possible to create such a PKCS12 file with cryptography or PyOpenSSL. The OpenSSL API used by both is a simplified API (`PKCS12_create`) which accepts one single friendly name (which is applied to the 'main' certificate). I guess we can return names we find for other certificates as well, but I'm not sure how useful this would be if it's not possible to create such a certificate without using third-party code.

(I'll start with tests, changelog and docs once it's clear how the interface should be.)

Reply:
I implemented what you suggested in 4dfb7a68f866ddd9579e12b5347b9f7cbbfafd15 together with an API to create such PKCS#12 files (`serialize_key_and_certificates_with_names()`), and added some basic tests.

Reply:
(I'm still hesitant at adding a `PKCS12` object at this step, since that kind of fixes the structure of PKCS#12 files that cryptography can handle for a longer time.)